File: 0605-can-place-flowers/0605-can-place-flowers.cpp
class Solution {
public:
   bool canPlaceFlowers(vector<int>& flowerbed, int n) {
      int count = 0, prev = 0;
      for (int x : flowerbed) {
         if (x == 1) {
            if (prev == 1) {
               count--;
            }
            prev = 1;
         } else {
            if (prev == 1) {
               prev = 0;
            } else {
               count++;
               prev = 1;
            }
         }
      }
      return count >= n; 
   }
};

File: 0605-can-place-flowers/0605-can-place-flowers.py

class Solution:
    def canPlaceFlowers(self, flowerbed: List[int], n: int) -> bool:
        for i in range(len(flowerbed)):
            if i == len(flowerbed) - 1:
                if flowerbed[i] == flowerbed[i-1] == 0:
                    n -= 1
                    flowerbed[i] = 1
            elif i == 0 and flowerbed[i] == flowerbed[i+1] == 0:
                n -= 1
                flowerbed[i] = 1
            elif flowerbed[i-1] == flowerbed[i] == flowerbed[i+1] == 0:
                n-=1
                flowerbed[i] = 1
            if n <= 0:
                return True
        return n <= 0
